src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -1,198 +1,199 @@
+	6: SEVERITY.MAJOR,
+}
+
+const MISSED_BROTHERHOOD_SEVERITY = {
+	1: SEVERITY.MEDIUM,
+	2: SEVERITY.MAJOR,
+}
+
+const MEDITATION_DURING_ROF_SEVERITY = {
+	1: SEVERITY.MEDIUM,
+	3: SEVERITY.MAJOR,
+}
+
 const EXPECTED = {
 	GCDS: 11,
-	BLITZES: 1,
 }
 
-const SUGGESTION_TIERS = {
-	1: SEVERITY.MINOR,
-	2: SEVERITY.MEDIUM,
-	5: SEVERITY.MAJOR,
-}
+class MasterfulBlitzEvaluator implements WindowEvaluator {
+	private _blitzSkills: Action[]
+	private _celestialRevolution: Action
+	private _masterfulBlitz: Action
 
-class Riddle {
-	data: Data
-	casts: Array<Events['action']>
-	start: number
-	end?: number
-	rushing: boolean = false
-
-	constructor(start: number, data: Data) {
-		this.data = data
-		this.start = start
-		this.casts = []
+	constructor(blitzSkills: Action[], celestialRevolution: Action, masterfulBlitz: Action) {
+		this._blitzSkills = blitzSkills
+		this._celestialRevolution = celestialRevolution
+		this._masterfulBlitz = masterfulBlitz
 	}
 
-	get gcds() {
-		return this.casts
-			.filter(event => this.data.getAction(event.action)?.onGcd ?? false)
+	suggest(windows: Array<HistoryEntry<EvaluatedAction[]>>) {
+		const celestialRevolutionUsages = windows.map(previousValue => previousValue.data)
+			.reduce((previousValue, currentValue) => previousValue.concat(currentValue))
+			.filter(value => value.action === this._celestialRevolution)
 			.length
+
+		if (celestialRevolutionUsages > 0) {
+			return new Suggestion({
+				icon: this._celestialRevolution.icon,
+				content: <Trans id="mnk.rof.suggestions.celestialrevolution.content">Avoid using <ActionLink
+					action="CELESTIAL_REVOLUTION"/>.</Trans>,
+				severity: SEVERITY.MAJOR,
+				why: <Trans id="mnk.rof.suggestions.celestialrevolution.why"><ActionLink
+					action="RISING_PHOENIX"/> and <ActionLink action="ELIXIR_FIELD"/> do more potency along with
+					AoE.</Trans>,
+			})
+		}
 	}
 
-	get chakras() {
-		return this.casts.filter(event => [this.data.actions.THE_FORBIDDEN_CHAKRA.id, this.data.actions.ENLIGHTENMENT.id].includes(event.action)).length
+	output(windows: Array<HistoryEntry<EvaluatedAction[]>>): EvaluationOutput {
+		return {
+			format: 'table',
+			header: {
+				header: <ActionLink showName={false} action={'MASTERFUL_BLITZ'}/>,
+				accessor: 'masterfulblitz',
+			},
+			rows: windows.map((window, i) => {
+				return {
+					actual: this.countBlitzSkills(window),
+					expected: this.numberOfBlitzExpectedInWindow(i),
+				}
+			}),
+		}
 	}
 
-	get meditations() {
-		return this.casts.filter(event => event.action === this.data.actions.MEDITATION.id).length
+	private countBlitzSkills(window: HistoryEntry<EvaluatedAction[]>): number {
+		return window.data.filter(value => this._blitzSkills.includes(value.action)).length
 	}
 
-	get stars() {
-		return this.casts.filter(event => event.action === this.data.actions.SIX_SIDED_STAR.id).length
+	private numberOfBlitzExpectedInWindow(windowIndex: number): number {
+		// 2 on odds and 1 on evens
+		return 2 - windowIndex % 2
```

Comment:
I'm not sure that actually solves it, since downtime exists. It's possible user could have enough charges of PB that they can do 2 blitzes in 2 consecutive windows I think. Might be worth a time check since last PB or something.

Reply:
Could I have something like: if the time between window 1 and window 2 is greater than 80 seconds (2 PB CDs), than I expect 2 blitzed in that window?